File: src/main/java/ru/arc/invest/BusinessManager.java
package ru.arc.invest;

import ru.arc.ARC;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

public class BusinessManager {

    private static Map<String, Business> businessMap = new ConcurrentHashMap<>();

    public static Business byName(String name){
        return businessMap.get(name);
    }

    public static void load(){

        Path path1 = Paths.get(ARC.plugin.getDataFolder().toString(), "investing", "businesses");
        if(!Files.exists(path1)){
            try {
                Files.createDirectories(path1);
                ARC.plugin.saveResource("investing/businesses/farm.yml", false);
            } catch (IOException e) {
                throw new RuntimeException(e);
            }
        }

        Path path2 = Paths.get(ARC.plugin.getDataFolder().toString(), "investing", "inventories");
        if(!Files.exists(path2)){
            try {
                Files.createDirectories(path2);
                ARC.plugin.saveResource("investing/inventories/farm.yml", false);
            } catch (IOException e) {
                throw new RuntimeException(e);
            }
        }

        try(var stream = Files.walk(Paths.get(ARC.plugin.getDataFolder().toString(), "investing", "businesses"), 3)){
            stream.filter(p -> p.toString().endsWith(".yml"))
                    .map(Path::getFileName)
                    .map(p -> p.toString().replace(".yml", ""))
                    .forEach(BusinessManager::crateFromName);
        } catch (Exception e){
            e.printStackTrace();
        }
    }


    private static void crateFromName(String name){
        Business business = new Business(name);
        business.load();
        businessMap.put(name, business);
        System.out.println("Business "+name+" loaded!");
    }


}

File: src/main/kotlin/ru/arc/invest/BusinessManager.kt
package ru.arc.invest

import ru.arc.ARC
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Paths
import java.util.concurrent.ConcurrentHashMap

object BusinessManager {

    private val businessMap = ConcurrentHashMap<String, Business>()

    @JvmStatic
    fun byName(name: String): Business? = businessMap[name]

    @JvmStatic
    fun load() {
        val path1 = Paths.get(ARC.instance.dataFolder.toString(), "investing", "businesses")
        if (!Files.exists(path1)) {
            try {
                Files.createDirectories(path1)
                ARC.instance.saveResource("investing/businesses/farm.yml", false)
            } catch (e: IOException) {
                throw RuntimeException(e)
            }
        }

        val path2 = Paths.get(ARC.instance.dataFolder.toString(), "investing", "inventories")
        if (!Files.exists(path2)) {
            try {
                Files.createDirectories(path2)
                ARC.instance.saveResource("investing/inventories/farm.yml", false)
            } catch (e: IOException) {
                throw RuntimeException(e)
            }
        }

        try {
            Files.walk(
                Paths.get(ARC.instance.dataFolder.toString(), "investing", "businesses"), 3
            ).use { stream ->
                stream.filter { it.toString().endsWith(".yml") }
                    .map { it.fileName }
                    .map { it.toString().replace(".yml", "") }
                    .forEach { createFromName(it) }
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    private fun createFromName(name: String) {
        val business = Business(name)
        business.load()
        businessMap[name] = business
        println("Business $name loaded!")
    }
}
